src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java

```diff
@@ -0,0 +1,172 @@
+        this.configModel = configModel;
+    }
+
+    @Subscribe
+    public void onDynamicConfigModelChanged(DynamicConfigModel dcm) {
+        this.dcm = dcm;
+
+        Settings settings = dcm.getDynamicOnBehalfOfSettings();
+
+        Boolean enabled = Boolean.parseBoolean(settings.get("enabled"));
+        String signingKey = settings.get("signing_key");
+        String encryptionKey = settings.get("encryption_key");
+
+        if (!Boolean.FALSE.equals(enabled) && signingKey != null && encryptionKey != null) {
+            this.vendor = new JwtVendor(settings, Optional.empty());
+        } else {
+            this.vendor = null;
+        }
+    }
+
+    public CreateOnBehalfOfTokenAction(final Settings settings, final ThreadPool threadPool, final ClusterService clusterService) {
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public String getName() {
+        return getClass().getSimpleName();
+    }
+
+    @Override
+    public List<Route> routes() {
+        return routes;
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        switch (request.method()) {
+            case POST:
+                return handlePost(request, client);
+            default:
+                throw new IllegalArgumentException(request.method() + " not supported");
+        }
+    }
+
+    private RestChannelConsumer handlePost(RestRequest request, NodeClient client) throws IOException {
+        return new RestChannelConsumer() {
+            @Override
+            public void accept(RestChannel channel) throws Exception {
+                final XContentBuilder builder = channel.newBuilder();
+                BytesRestResponse response;
+                try {
+                    if (vendor == null) {
+                        channel.sendResponse(
+                            new BytesRestResponse(
+                                RestStatus.SERVICE_UNAVAILABLE,
+                                "on_behalf_of is either disabled or the configuration is invalid"
+                            )
+                        );
+                        return;
+                    }
+
+                    final String clusterIdentifier = clusterService.getClusterName().value();
+
+                    final Map<String, Object> requestBody = request.contentOrSourceParamParser().map();
+                    final String description = (String) requestBody.getOrDefault("description", null);
+
+                    final Integer tokenDuration = Optional.ofNullable(requestBody.get("duration"))
+                        .map(value -> (String) value)
+                        .map(Integer::parseInt)
+                        .map(value -> Math.min(value, OBO_MAX_EXPIRY_SECONDS)) // Max duration is 10 minutes
+                        .orElse(OBO_DEFAULT_EXPIRY_SECONDS); // Fallback to default of 5 minutes;
```

Comment:
Don't include 'values' in comments since they can easily become out of date.

```suggestion
                        .orElse(OBO_DEFAULT_EXPIRY_SECONDS); // If not provided fallback to default
```

Reply:
fixed